specification/monitor/resource-manager/microsoft.insights/preview/2018-06-01-preview/guestDiagnosticSettings_API.json

```diff
@@ -0,0 +1,580 @@
+                },
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The request succeeded.",
+                        "schema": {
+                            "$ref": "#/definitions/GuestDiagnosticSettingsList"
+                        }
+                    },
+                    "default": {
+                        "description": "An error occurred and the list of guest diagnostic settings could not be retrieved.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
+        }
+    },    
+    "definitions": {
+        "ErrorResponse": {
+            "description": "Describes the format of Error response.",
+            "type": "object",
+            "properties": {
+                "code": {
+                    "description": "Error code",
+                    "type": "string"
+                },
+                "message": {
+                    "description": "Error message indicating why the operation failed.",
+                    "type": "string"
+                }
+            }
+        },
+        "GuestDiagnosticSettingsList": {
+            "description": "A list of guest diagnostic settings.",
+            "properties": {
+                "value": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/GuestDiagnosticSettingsResource"
+                    },
+                    "description": "The list of guest diagnostic settings."
+                },
+                "nextLink": {
+                    "type": "string",
+                    "description": "Provides the link to retrieve the next set of elements."
+                }
+            }
+        },
+        "GuestDiagnosticSettingsResource": {
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Resource"
+                }
+            ],
+            "required": [
+                "properties"
+            ],
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "$ref": "#/definitions/DiagnosticSettings",
+                    "description": "The diagnostic settings to be applied to azure resources."
+                }
+            },
+            "description": "Virtual machine guest diagnostics settings resource."
+        },
+        "DiagnosticSettings": {
```

Comment:
This model name is incompatible with an already existing model name:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/microsoft.insights/preview/2017-05-01-preview/diagnosticsSettings_API.json#L277

Do you intend to generate both specs into the same SDK? If so, this will need to be renamed (e.g. "GuestDiagnosticSettings"). Otherwise if this spec is intended to replace diagnosticSettings_API then it should be okay.